app/controllers/foods_controller.rb

```diff
@@ -1,11 +1,38 @@
 class FoodsController < ApplicationController
-  def index; end
+  before_action :authenticate_user!
 
-  def show; end
+  def index
+    @user = current_user
+    @foods = Food.all
+  end
 
-  def new; end
+  def show
+    @food = Food.find(params[:id])
+  end
 
-  def create; end
+  def new
+    @food = current_user.foods.new
+  end
 
-  def destroy; end
+  def create
+    @food = current_user.foods.create(food_params)
+
+    if @food.save
+      redirect_to new_food_path
```

Comment:
After creating a new food, the method should redirect back to the foods list not the new food form.
You can do it using: `redirect_to foods_path`.

Also the method should have some flash messages to interact with the user, you can check the inventories controller for more information.

Reply:
Already done